packages/augur-sdk/src/state/db/MarketDB.ts

```diff
@@ -36,43 +52,41 @@ interface LiquidityResults {
   [liquidity: number]: number;
 }
 
-// because flexsearch is a UMD type lib
-import FlexSearch = require("flexsearch");
-
 /**
  * Market specific derived DB intended for filtering purposes
  */
 export class MarketDB extends DerivedDB {
   protected augur: Augur;
   private readonly events = new Subscriptions(augurEmitter);
-  private flexSearch?: FlexSearch;
+  private flexSearchIndex?: Index<MarketFields>;
   readonly liquiditySpreads = [10, 15, 20, 100];
 
   constructor(db: DB, networkId: number, augur: Augur) {
     super(db, networkId, "Markets", ["MarketCreated", "MarketVolumeChanged", "MarketOIChanged"], ["market"]);
-    
+
     this.augur = augur;
 
     this.events.subscribe('DerivedDB:updated:CurrentOrders', this.syncOrderBooks);
 
-    this.flexSearch = new FlexSearch({
-      doc: {
-        id: "id",
-        start: "start",
-        end: "end",
-        field: [
-          "market",
-          "category1",
-          "category2",
-          "category3",
-          "description",
-          "longDescription",
-          "resolutionSource",
-          "_scalarDenomination",
-          "tags",
-        ],
-      },
-    });
+    this.flexSearchIndex = flexSearch.create(
+      {
+        doc: {
+          id: "id",
+          start: "start",
+          end: "end",
+          field: [
+            "market",
+            "category1",
+            "category2",
+            "category3",
```

Comment:
what if category3 is empty will that mess up flex search?

Reply:
If category3 is specified but a market does not have it, should it be included in the results?

Reply:
I might not be understanding the code above. Does this above just create the index for search? not actually add data?

Reply:
gotcha on the above. 👍

Reply:
Correct, this just creates the index for the field.